### PR TITLE
Adding more XML doc comments; a few breaking renames

### LINKED ISSUE #588: Rename CommandLineBuilder.UseMiddleware extension method
You can call UseMiddleware multiple times and each one will be added to the pipeline. The name implies if you called it twice only the last call would be applied. It is a fluent wrapper over the internal void CommandLineBuilder.AddMiddleware.

### PR DESCRIPTION
**Pull request by jonsequitur**
This PR also contains the following breaking renames: 

* `AliasedSet.Contains` is now `AliasedSet.ContainsAlias`
* `CommandLineBuilder.UseResources` is now `CommandLineBuilder.UseLocalizationResources`
* `CommandLineBuilder.UseMiddleware` is now `CommandLineBuilder.AddMiddleware` (fixes #588)
* Removed `CommandLineBuilder.UseHelp` overloads that can customize the `HelpBuilder`.
* Removed the ability to set exit codes via `UseDebugDirective` and `UseVersionOption`
* Removed `ModelBinder.BindConstructorArgumentFromValue`

### CODE REVIEW

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 1]
```diff
@@ -48,6 +48,13 @@ public static class ArgumentExtensions
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only the specified values, and to suggest them as command line completions.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <param name="values">The values that are allowed for the argument.</param>
+        /// <typeparam name="TArgument">The type of the argument.</typeparam>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:57]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 2]
```diff
@@ -59,6 +66,11 @@ public static class ArgumentExtensions
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing file.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:73]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 3]
```diff
@@ -70,6 +82,11 @@ public static Argument<FileInfo> ExistingOnly(this Argument<FileInfo> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing directory.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:89]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 4]
```diff
@@ -81,6 +98,11 @@ public static Argument<DirectoryInfo> ExistingOnly(this Argument<DirectoryInfo>
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing file or directory.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:105]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 5]
```diff
@@ -92,6 +114,11 @@ public static Argument<FileSystemInfo> ExistingOnly(this Argument<FileSystemInfo
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to a existing files or directories.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:121]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 6]
```diff
@@ -126,6 +153,11 @@ public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values representing legal file paths.
+        /// </summary>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:160]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 7]
```diff
@@ -154,6 +186,12 @@ public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values representing legal file names.
+        /// </summary>
+        /// <remarks>A parse error will result, for example, if file path separators are found in the parsed value.</remarks>
+        /// <param name="argument">The argument to configure.</param>
+        /// <returns>The argument being extended.</returns>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:194]:
```suggestion
        /// <returns>The configured argument.</returns>
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 8]
```diff
@@ -48,6 +48,13 @@ public static class ArgumentExtensions
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only the specified values, and to suggest them as command line completions.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:52]:
```suggestion
        /// Configures an argument to only accept the specified values, and to suggest them as command line completions.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 9]
```diff
@@ -59,6 +66,11 @@ public static class ArgumentExtensions
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing file.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:70]:
```suggestion
        /// Configures an argument to only accept values corresponding to an existing file.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 10]
```diff
@@ -70,6 +82,11 @@ public static Argument<FileInfo> ExistingOnly(this Argument<FileInfo> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing directory.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:86]:
```suggestion
        /// Configures an argument to only accept values corresponding to an existing directory.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 11]
```diff
@@ -81,6 +98,11 @@ public static Argument<DirectoryInfo> ExistingOnly(this Argument<DirectoryInfo>
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to an existing file or directory.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:102]:
```suggestion
        /// Configures an argument to only accept values corresponding to an existing file or directory.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 12]
```diff
@@ -92,6 +114,11 @@ public static Argument<FileSystemInfo> ExistingOnly(this Argument<FileSystemInfo
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values corresponding to a existing files or directories.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:118]:
```suggestion
        /// Configures an argument to only accept values corresponding to a existing files or directories.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 13]
```diff
@@ -126,6 +153,11 @@ public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values representing legal file paths.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:157]:
```suggestion
        /// Configures an argument to only accept values representing legal file paths.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 14]
```diff
@@ -154,6 +186,12 @@ public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
             return argument;
         }
 
+        /// <summary>
+        /// Configures an argument to accept only values representing legal file names.
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:190]:
```suggestion
        /// Configures an argument to only accept values representing legal file names.
```

File: src/System.CommandLine/ArgumentExtensions.cs [hunk 15]
```diff
@@ -178,7 +216,14 @@ public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
...
             return argument;
         }
-
+        
+        /// <summary>
+        /// Parses a command line string value using an argument.
+        /// </summary>
+        /// <remarks>The command line string input will be split into tokens as if it had been passed on the command line.</remarks>
+        /// <param name="argument">The argument to use to parse the command line input.</param>
+        /// <param name="commandLine">A command line string input.</param>
```
Comment (Keboo) [inline src/System.CommandLine/ArgumentExtensions.cs:225]:
```suggestion
        /// <param name="commandLine">A command line string to parse.</param>
```

File: src/System.CommandLine/CommandExtensions.cs [hunk 16]
```diff
@@ -15,6 +15,13 @@ namespace System.CommandLine
     /// </summary>
     public static class CommandExtensions
     {
+        /// <summary>
+        /// Parses and invokes a command.
+        /// </summary>
+        /// <param name="command">The command to invoke.</param>
+        /// <param name="args">The arguments to pass.</param>
```
Comment (Keboo) [inline src/System.CommandLine/CommandExtensions.cs:22]:
```suggestion
        /// <param name="args">The arguments to parse.</param>
```

File: src/System.CommandLine/CommandExtensions.cs [hunk 17]
```diff
@@ -23,12 +30,27 @@ public static class CommandExtensions
             return GetInvocationPipeline(command, args).Invoke(console);
         }
 
+        /// <summary>
+        /// Parses and invokes a command.
+        /// </summary>
+        /// <remarks>The command line string input will be split into tokens as if it had been passed on the command line.</remarks>
+        /// <param name="command">The command to invoke.</param>
+        /// <param name="commandLine">The command line to pass.</param>
```
Comment (Keboo) [inline src/System.CommandLine/CommandExtensions.cs:38]:
```suggestion
        /// <param name="commandLine">The command line to parse.</param>
```

File: src/System.CommandLine/CommandExtensions.cs [hunk 18]
```diff
@@ -23,12 +30,27 @@ public static class CommandExtensions
...
             this Command command,
             string commandLine,
             IConsole? console = null) =>
             command.Invoke(CommandLineStringSplitter.Instance.Split(commandLine).ToArray(), console);
 
+        /// <summary>
+        /// Parses and invokes a command.
+        /// </summary>
+        /// <param name="command">The command to invoke.</param>
+        /// <param name="args">The arguments to pass.</param>
```
Comment (Keboo) [inline src/System.CommandLine/CommandExtensions.cs:51]:
```suggestion
        /// <param name="args">The arguments to parse.</param>
```

File: src/System.CommandLine/CommandExtensions.cs [hunk 19]
```diff
@@ -37,6 +59,14 @@ public static class CommandExtensions
             return await GetInvocationPipeline(command, args).InvokeAsync(console);
         }
 
+        /// <summary>
+        /// Parses and invokes a command.
+        /// </summary>
+        /// <remarks>The command line string input will be split into tokens as if it had been passed on the command line.</remarks>
+        /// <param name="command">The command to invoke.</param>
+        /// <param name="commandLine">The command line to pass.</param>
```
Comment (Keboo) [inline src/System.CommandLine/CommandExtensions.cs:67]:
```suggestion
        /// <param name="commandLine">The command line to parse.</param>
```

File: src/System.CommandLine/ConsoleExtensions.cs [hunk 20]
```diff
@@ -10,10 +10,20 @@ namespace System.CommandLine
...
+        /// Writes the current string value to the standard output stream.
+        /// </summary>
+        /// <param name="console">The console to write to.</param>
+        /// <param name="value">The value to write.</param>
+        public static void Write(this IConsole console, string value) =>
             console.Out.Write(value);
 
-        public static void WriteLine(this IConsole console, string value) => 
+        /// <summary>
+        /// Writes the current string value, followed by the current line terminator, to the standard output stream.
```
Comment (Keboo) [inline src/System.CommandLine/ConsoleExtensions.cs:22]:
```suggestion
        /// Writes the current string value, followed by the current environment's line terminator, to the standard output stream.
```

File: src/System.CommandLine/ArgumentExtensions.cs (hunk 8 above)
Comment (jonsequitur) [reply src/System.CommandLine/ArgumentExtensions.cs:52]: I shall not accept your split infinitive, sir!

**Pull request merged**